File: code/ch09/9.1.2.1.container_with_most_water.py
class Solution :
    def maxArea(self, height:List[int])->int:
        res, area = 0, 0
        for i in range(0, len(height), 1):
            fo j in range(i + 1, len(height), 1):
                area = (j - i) * min(height[i], height[j])
                if area > res:
                    res = area

        return res
    

File: code/ch09/9.1.2.1.container_with_most_water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int res = 0;
        int area = 0;
        for (int i = 0; i < height.size() - 1; i++)
            for (int j = i + 1; j < height.size(); j++) {
                area = (j - i) * min(height[i], height[j]);
                if (area > res)
                    res = area;
            }
        
        return res;
    }
};
